### PR TITLE
MAINT-21066: Make sure that news activities notifications aren't duplicated.

### PR DESCRIPTION
**Pull request by mdiha**
I made sure to avoid broadcasting some news activity notifications that are already handled by news module.

### CODE REVIEW

File: component/notification/src/main/java/org/exoplatform/social/notification/Utils.java [hunk 1]
```diff
@@ -63,6 +64,11 @@
       "(?:[\\/|\\?|\\#].*)?$");                                                               // path and query
   
   private static final String styleCSS = " style=\"color: #2f5e92; text-decoration: none;\"";
+
+  /**
+   * Exo property name used for disable news activity notifications
+   */
+  private static final String NEWS_ACTIVITY_NOTIFICATIONS_PROPERTY_NAME = "exo\\.notifications\\.activity-type.news-shared_news\\.enabled";
```
Comment (ahamdi) [inline component/notification/src/main/java/org/exoplatform/social/notification/Utils.java]:
```suggestion
  private static final String NEWS_ACTIVITY_NOTIFICATIONS_PROPERTY_NAME = "exo\\.notifications\\.activity-type\\.enabled";
```

File: component/notification/src/main/java/org/exoplatform/social/notification/Utils.java [hunk 2]
```diff
@@ -291,4 +297,8 @@ public static ActivityManager getActivityManager() {
   public static RelationshipManager getRelationshipManager() {
     return getService(RelationshipManager.class);
   }
+
+  public static boolean isNewsActivityNotificationsEnabled() {
+    return Boolean.parseBoolean(PropertyManager.getProperty(NEWS_ACTIVITY_NOTIFICATIONS_PROPERTY_NAME));
+  }
```
Comment (ahamdi) [inline component/notification/src/main/java/org/exoplatform/social/notification/Utils.java]: Please update the PR based on the logic I proposed

```suggestion
  /**
   * Checks if the notification is enabled for activities of type activityType
   * @param activityType type of activity to check if their notification is enabled
   * @return true if the notification is enabled for this Activity Type
   */
  public static boolean isActivityNotificationsEnabled(String activityType) {
    String disabledNotifications = PropertyManager.getProperty(NEWS_ACTIVITY_NOTIFICATIONS_PROPERTY_NAME);
    if(StringUtils.isNotBlank(disabledNotifications)) {
      String[] activityTypes = disabledNotifications.split(",");
      return ! Arrays.asList(activityTypes).contains(activityType);
    }
    return true;
  }
```

**Review by ahamdi — CHANGES_REQUESTED** [review]

File: component/notification/src/main/java/org/exoplatform/social/notification/Utils.java (hunk 2 above)
Comment (mdiha) [reply component/notification/src/main/java/org/exoplatform/social/notification/Utils.java]: Okay

File: component/notification/src/main/java/org/exoplatform/social/notification/Utils.java (hunk 1 above)
Comment (mdiha) [reply component/notification/src/main/java/org/exoplatform/social/notification/Utils.java]: Okay

File: component/notification/src/main/java/org/exoplatform/social/notification/plugin/ActivityCommentPlugin.java [hunk 3]
```diff
@@ -87,7 +86,12 @@ public boolean isValid(NotificationContext ctx) {
     ExoSocialActivity comment = ctx.value(SocialNotificationUtils.ACTIVITY);
     ExoSocialActivity activity = Utils.getActivityManager().getParentActivity(comment);
 
-    if(isSubComment && comment.getParentCommentId() == null) {
+    if (StringUtils.equals(activity.getType(), "news") && !Utils.isActivityNotificationsEnabled("news") ||
+            StringUtils.equals(activity.getType(), "shared_news") && !Utils.isActivityNotificationsEnabled("shared_news")) {
+      return false;
```
Comment (ahamdi) [inline component/notification/src/main/java/org/exoplatform/social/notification/plugin/ActivityCommentPlugin.java]: No need to hardcode news and shared_news, just use the new API :+1: 
```suggestion
    if (!Utils.isActivityNotificationsEnabled(activity.getType())) {
      return false;
```

File: component/notification/src/main/java/org/exoplatform/social/notification/plugin/LikePlugin.java [hunk 4]
```diff
@@ -62,8 +63,14 @@ public NotificationInfo makeNotification(NotificationContext ctx) {
   @Override
   public boolean isValid(NotificationContext ctx) {
     ExoSocialActivity activity = ctx.value(SocialNotificationUtils.ACTIVITY);
+
+    if (StringUtils.equals(activity.getType(), "news") && !Utils.isActivityNotificationsEnabled("news") ||
+            StringUtils.equals(activity.getType(), "shared_news") && !Utils.isActivityNotificationsEnabled("shared_news")) {
+      return false;
+    }
```
Comment (ahamdi) [inline component/notification/src/main/java/org/exoplatform/social/notification/plugin/LikePlugin.java]: The same as above

File: component/notification/src/main/java/org/exoplatform/social/notification/plugin/PostActivitySpaceStreamPlugin.java [hunk 5]
```diff
@@ -62,12 +63,18 @@ public NotificationInfo makeNotification(NotificationContext ctx) {
   @Override
   public boolean isValid(NotificationContext ctx) {
     ExoSocialActivity activity = ctx.value(SocialNotificationUtils.ACTIVITY);
+
+    if (StringUtils.equals(activity.getType(), "news") && !Utils.isActivityNotificationsEnabled("news") ||
+            StringUtils.equals(activity.getType(), "shared_news") && !Utils.isActivityNotificationsEnabled("shared_news")) {
+      return false;
+    }
```
Comment (ahamdi) [inline component/notification/src/main/java/org/exoplatform/social/notification/plugin/PostActivitySpaceStreamPlugin.java]: The same remark

**Review by ahamdi — CHANGES_REQUESTED** [review]

File: component/notification/src/main/java/org/exoplatform/social/notification/plugin/ActivityCommentPlugin.java (hunk 3 above)
Comment (mdiha) [reply component/notification/src/main/java/org/exoplatform/social/notification/plugin/ActivityCommentPlugin.java]: Yes you are right (y)

File: component/notification/src/main/java/org/exoplatform/social/notification/plugin/LikePlugin.java (hunk 4 above)
Comment (mdiha) [reply component/notification/src/main/java/org/exoplatform/social/notification/plugin/LikePlugin.java]: Done

File: component/notification/src/main/java/org/exoplatform/social/notification/plugin/PostActivitySpaceStreamPlugin.java (hunk 5 above)
Comment (mdiha) [reply component/notification/src/main/java/org/exoplatform/social/notification/plugin/PostActivitySpaceStreamPlugin.java]: Done

**Review by ahamdi — DISMISSED** [review]
LGTM
Let's document this @mzorai @mdiha

**Review by ahamdi — APPROVED** [review]

**Pull request merged**